.github/workflows/check_precompiled.yml

```diff
@@ -23,50 +26,65 @@ jobs:
           path: pico-sdk
           submodules: 'true'
 
-      - name: Create copies of the precompiled binaries
-        run: |
-          cp enc_bootloader/enc_bootloader.elf enc_bootloader/enc_bootloader.elf.orig
-          cp enc_bootloader/enc_bootloader_mbedtls.elf enc_bootloader/enc_bootloader_mbedtls.elf.orig
-          cp picoboot_flash_id/flash_id.bin picoboot_flash_id/flash_id.bin.orig
-          cp xip_ram_perms/xip_ram_perms.elf xip_ram_perms/xip_ram_perms.elf.orig
-
       - name: Build and Install
         run: |
           cmake -S . -B build -G "Ninja" -D PICO_SDK_PATH="${{ github.workspace }}/pico-sdk" -D USE_PRECOMPILED=FALSE
           cmake --build build
           sudo cmake --install build
 
-      - name: Check if precompiled binaries are up to date
-        env:
-          ALL_FINE: 1
+      - name: Check precompiled binaries have been updated
         run: |
-          # Compare the files
-          if ! cmp -s enc_bootloader/enc_bootloader.elf enc_bootloader/enc_bootloader.elf.orig; then
-            echo "Error: enc_bootloader.elf is out of date"
-            ALL_FINE=0
-          fi
+          updated_files=$(curl -s -u "${{ github.repository_owner }}":"${{ github.token }}" -H "Accept: application/vnd.github.v3+json" "${{ github.event.pull_request._links.self.href }}/files")
+          updated_files=$(jq -r '.[] | .filename' <<<"$updated_files")
 
-          if ! cmp -s enc_bootloader/enc_bootloader_mbedtls.elf enc_bootloader/enc_bootloader_mbedtls.elf.orig; then
-            echo "Error: enc_bootloader_mbedtls.elf is out of date"
-            ALL_FINE=0
-          fi
+          echo "Updated files: $updated_files"
 
-          if ! cmp -s picoboot_flash_id/flash_id.bin picoboot_flash_id/flash_id.bin.orig; then
-            echo "Error: flash_id.bin is out of date"
-            ALL_FINE=0
-          fi
+          # enc_bootloader
+          enc_bootloader_elf_not_updated=$(! echo "$updated_files" | grep -q "enc_bootloader/enc_bootloader.elf")
```

Comment:
Might be worth tightening up your grep regexes slightly?
```console
$ echo "FOOenc_bootloader/enc_booloaderXelfBAR" | grep "enc_bootloader/enc_booloader.elf"
FOOenc_bootloader/enc_booloaderXelfBAR
```
:wink:

Reply:
I've anchored the starts & ends